### ISSUE TITLE
Emit method overload symbol in Java mtags

### ISSUE THREAD

**Issue by olafurpg** [scalameta/metals#281]
Java mtags currently emits field symbols for methods.

```java
void foo(int a, String b) 
// field: foo.
// method foo(Int, String)
```
it would be nice if java mtags emitted correct method symbols so it's possible to goto definition to the precise overload of a method.

The specification for java method symbol syntax is documented here https://github.com/scalameta/scalameta/blob/master/semanticdb/semanticdb3/semanticdb3.md#java-symbol

The place to fix this issue is here https://github.com/scalameta/metals/blob/905bf282299f7c136b40173103cf367073dc55f8/metals/src/main/scala/scala/meta/metals/mtags/JavaMtags.scala
and to test the fix

https://github.com/scalameta/metals/blob/905bf282299f7c136b40173103cf367073dc55f8/metals/src/test/scala/tests/mtags/JavaMtagsTest.scala

**Comment by scalavision** [comment]
I have been working a bit on this issue:

https://github.com/scalavision/metals/tree/3.7

This is still missing:
* support for Arrays
* More tests needed

**Comment by olafurpg** [comment]
Fixed in #292, Java mtags emits correct method overloads including for overloaded methods between non-static and static methods (which requires a special case in the SemanticDB spec).
